roles/confluent.common/tasks/main.yml

```
@@ -31,6 +31,49 @@
     state: absent
     regexp: '^Defaults\s+requiretty.*$'
 
+## Process Archive File deployments
+- name: Create Confluent Platform install directory
+  file:
+    path: "{{archive_destination_path}}"
+    state: directory
+    mode: 0755
+  when: archive_file_enabled|bool
+
+- name: Test for local Confluent Platform archive
+  stat:
+    path: "{{ confluent_archive_file_path }}"
+  register: local_confluent_archive
+
+- name: Create a temp directory to stage the Confluent Platform archive
+  tempfile:
+    state: directory
+    suffix: staging
+  register: archive_staging_tmp
+  when: archive_file_enabled|bool
+
+- name: Copy local Confluent Platform archive to a temporary directory
+  copy:
+    src: "{{ confluent_archive_file_path }}"
+    remote_src: yes
+    dest: "{{archive_staging_tmp.path}}"
+  when: local_confluent_archive.stat.exists == True
+
+- name: Download Confluent Platform archive
+  get_url:
+    url: "{{ confluent_common_repository_archive_url }}"
+    dest: "{{archive_staging_tmp.path}}"
+  when:
+    - archive_file_enabled|bool
+    - local_confluent_archive.stat.exists == False
+
+- name: Expand Confluent Platform archive
+  unarchive:
+    src: "{{archive_staging_tmp.path}}/{{confluent_archive_file_name}}"
+    remote_src: yes
+    dest: "{{archive_destination_path}}"
+    mode: 0755
+  when: archive_file_enabled|bool
```

```suggestion
- name: Download Confluent Platform Archive
  unarchive:
    src: "{{ confluent_common_repository_archive_url }}"
    remote_src: yes
    dest: "{{archive_destination_path}}"
    mode: 0755
    creates: "{{ archive_destination_path }}"
  when: archive_file_enabled|bool
```

So I think the unarchive module can handle the "if exists" with `creates`. it can also handle downloading from a url as well
https://docs.ansible.com/ansible/latest/modules/unarchive_module.html

Can you please validate the suggested change above? 

I may be mixing up the confluent_archive_destination_path with archive_destination_path vars.

---

The concept here was to look locally for an archive and if it exists then push it to the staging/temp directory. If a local file isn't found then perform the download into the staging/temp directory. After the file is sourced from somewhere then it will be unarchived into the target installation directory. I'll look into a way to check if the installation directory exists and skip these steps, which I think is your intent.

---

I just pushed commit 99e9050a simplifies the overall logic to the following

1. Test for a local archive
1. If it exists then unarchive it directly into the target directory unless the target directory exists
1. Unarchive the remote archive URL directly into the target directory unless the target directory exists

I also skip modifications to the yum/apt repos (adding the confluent repos) when in archive file mode because it's not needed.

---

OOOO i misinterpreted! So you can either have the tar on the ansible control node or it will get the url! 
Sorry :)

It still looks like duplicating code to me though. Lets resolve this one and make a new comment